blog/views.py

```diff
@@ -1,3 +1,16 @@
 from django.shortcuts import render
 
 # Create your views here.
+from django.views import generic
+from .models import Post
+
+class PostList(generic.ListView):
+    queryset = Post.objects.filter(status=1).order_by('-created_on')
+    template_name = 'index.html'
+
+class PostDetail(generic.DetailView):
+    model = Post
+    template_name = 'post_detail.html'
+    
+def index(request=None):
```

Comment:
You might want to use a `TemplateView` here, and in fact, if you do then you can just reference the TemplateView in the `blog.urls` file directly without having to explicitly subclass it here.